packages/slice-machine/src/hooks/useLocalStorageItem.ts

```diff
@@ -0,0 +1,62 @@
+import { Dispatch, SetStateAction, useMemo, useSyncExternalStore } from "react";
+
+export type UseLocalStorageItemInfo = {
+  /** Whether the item is unset in localStorage and using the defaultValue as a fallback. */
+  isUnset: boolean;
+};
+
+export type UseLocalStorageReturnType<T> = [
+  T,
+  Dispatch<SetStateAction<T>>,
+  UseLocalStorageItemInfo,
+];
+
+const SLICE_MACHINE_STORAGE_PREFIX = "slice-machine";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+const isFunction = (value: any): value is (...args: any[]) => any => {
```

Comment:
❓ #ask: wouldn't type `unknown` be suitable here instead of any?

Reply:
Using `unknown` doesn't allow the type of the value to be inferred correctly

With `unknown`
![Screenshot 2024-07-16 at 16 42 35](https://github.com/user-attachments/assets/9a01b8a9-ba3c-46ad-bdb7-fabc77c3bdfb)

With `any`
![Screenshot 2024-07-16 at 16 40 24](https://github.com/user-attachments/assets/2704d8b3-f1a2-4797-b3b5-010f71419eed)

Reply:
thanks for explanation!

Reply:
More context: Doing an inline `typeof value === 'function'` should suffice here, but I think there's a bug in this typescript version, hence the function with a type guard. I will try to look into that in the future and propose to update it, as we're a bit behind on 4.9.5